### PR TITLE
[Jonathan Iheme] - Sage - Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2

### PR DESCRIPTION
**Pull request by 4undRaiser**
### Thanks for submitting this tutorial for Celo Sage

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines) and reviewed the [Sample Tutorial Structure](https://docs.celo.org/community/celo-sage/tutorial-structure).

Please confirm the following before submitting your tutorial for review -

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the tutorial template structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've run the tutorial to test it and the code works.

> NOTE: We cannot review your Pull Request if it does not meet the standard outlined above.
> Reviewers - @joenyzio & @viral-sangani

### CODE REVIEW

File: blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md [hunk 1]
```diff
@@ -0,0 +1,797 @@
...
+import { useState, useEffect, useCallback } from "react";
+import Web3 from "web3";
+import { newKitFromWeb3 } from "@celo/contractkit";
+import freelancer from "./contracts/freelancer.abi.json";
+import IERC from "./contracts/IERC.abi.json";
+import BigNumber from "bignumber.js";
+
+```
+
+#### Declaring constants and initializing state:
```
Comment (Julius170) [inline blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md:249]: can you make this a double hashtag or triple at most or maybe just use a bold text

File: blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md [hunk 2]
```diff
@@ -0,0 +1,797 @@
...
+  const [contract, setcontract] = useState(null);
+  const [address, setAddress] = useState(null);
+  const [kit, setKit] = useState(null);
+  const [cUSDBalance, setcUSDBalance] = useState(0);
+  const [freelancers, setFreelancers] = useState([]);
+  // ...
+}
+```
+
+#### Connecting to the user's wallet and get user's balance:
```
Comment (Julius170) [inline blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md:268]: can you make this a double hashtag or triple at most or maybe just use a bold text

File: blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md [hunk 3]
```diff
@@ -0,0 +1,797 @@
...
+      console.log(error);
+    }
+  }, [address, kit]);
+```
+
+We create a `connectToWallet` function that checks if the Celo extension is available, then connects to the user's wallet, and sets the user's default account.
+
+We also create a `getBalance` function using useCallback to fetch the user's cUSD balance and initialize the freelancer contract.
+
+#### Funtions to interact with the smart contract:
```
Comment (Julius170) [inline blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md:310]: can you make this a double hashtag or triple at most or maybe just use a bold text

File: blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md [hunk 4]
```diff
@@ -0,0 +1,797 @@
...
+      alert("you have successfully sent cusd to this user");
+    } catch (error) {
+      alert(error);
+    }
+  };
+```
+
+We define an async function called `hireFreelancer` that calculates the total payment based on the freelancer's hourly rate and the number of hours worked, approves the payment in cUSD, and calls the `hireFreelancerHourly` function from the smart contract.
+
+#### useEffect hooks:
```
Comment (Julius170) [inline blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md:434]: can you make this a double hashtag or triple at most or maybe just use a bold text

File: blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md [hunk 5]
```diff
@@ -0,0 +1,797 @@
...
+  useEffect(() => {
+    if (contract) {
+      getFreelancers();
+    }
+  }, [contract, getFreelancers]);
+```
+
+We use three `useEffect` hooks to automatically connect to the wallet, fetch the user's cUSD balance, and fetch the list of freelancers when the component is mounted or when any of the relevant state variables change.
+
+#### Rendering the components:
```
Comment (Julius170) [inline blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md:456]: can you make this a double hashtag or triple at most or maybe just use a bold text

**Review by Julius170 — COMMENTED** [review]
Hi @4undRaiser I left some comments about the 4 hashtags, you can chem them out

File: blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md [hunk 6]
```diff
@@ -0,0 +1,795 @@
+---
+title: Build A Decentralized Freelancer Marketplace On The Celo Blockchain Part 2
+description: In this tutorial, we will build the front end for the part 1 where we built the smart contract for a freelancer marketplace where people can find freelancers for their projects
+authors:
+  - name: ✍️ Jonathan Iheme
+    url: https://github.com/4undRaiser
+    image_url: https://avatars.githubusercontent.com/u/87926451?s=96&v=4
+tags: [celosage, solidity, celo, intermediate]
+hide_table_of_contents: true
+slug: "/tutorials/build-a-decentralized-freelancer-marketplace-on-the-celo-blockchain-part-2"
```
Comment (Julius170) [inline blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md:10]: remove the quotes at the beginning and the end of the slug value

**Review by Ishanpathak1 — APPROVED** [review]
Approved @developerkunal

**Pull request merged**